controllers/operatorpolicy_controller.go

```diff
@@ -60,180 +78,193 @@ var _ reconcile.Reconciler = &OperatorPolicyReconciler{}
 func (r *OperatorPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	policy := &policyv1beta1.OperatorPolicy{}
 
+	watcher := depclient.ObjectIdentifier{
+		Group:     operatorv1.GroupVersion.Group,
+		Version:   operatorv1.GroupVersion.Version,
+		Kind:      "OperatorPolicy",
+		Namespace: req.Namespace,
+		Name:      req.Name,
+	}
+
+	// Get the applied OperatorPolicy
 	err := r.Get(ctx, req.NamespacedName, policy)
 	if err != nil {
 		if errors.IsNotFound(err) {
-			OpLog.Info("Operator policy could not be found")
+			OpLog.Info("Operator policy could not be found", "name", req.Name, "namespace", req.Namespace)
+
+			err = r.DynamicWatcher.RemoveWatcher(watcher)
+			if err != nil {
+				OpLog.Error(err, "Error updating dependency watcher. Ignoring the failure.")
+			}
 
 			return reconcile.Result{}, nil
 		}
-	}
 
-	return reconcile.Result{}, nil
-}
+		OpLog.Error(err, "Failed to get operator policy")
```

Comment:
Should this log have some context or is that already set elsewhere?

Reply:
Good note. I want to try using the logger ctrl-runtime now puts into the context ([FromContext](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.14.6/pkg/log#FromContext)); I think that would include this information.